src/service/chat.js

```diff
@@ -161,11 +161,55 @@ const getOrCreateAnonymousChannel = async (userId) => {
   }
 };
 
+const moveChatToSigned = async ({oldChannelId, targetUserId, source}) => {
+  try {
+    const response = await api.post('/chat/move-to-sign', {oldChannelId, targetUserId, source});
+    if (response.status === 200) {
+      return Promise.resolve(response.data);
+    }
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
+};
+
+const moveChatToAnon = async ({
+  anon_user_info_color_code,
+  anon_user_info_color_name,
+  anon_user_info_emoji_code,
+  anon_user_info_emoji_name,
+  oldChannelId,
+  targetUserId,
+  source
+}) => {
+  try {
+    const response = await anonymousApi.post('/chat/move-to-anon', {
+      anon_user_info_color_code,
+      anon_user_info_color_name,
+      anon_user_info_emoji_code,
+      anon_user_info_emoji_name,
+      oldChannelId,
+      targetUserId,
+      source
+    });
+    if (response.status === 200) {
+      return Promise.resolve(response.data);
+    }
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar issues are present in the `moveChatToAnon` function. The error handling can be improved by rejecting with a specific error message and logging the error for debugging purposes.

```diff
-    if (response.status === 200) {
-      return Promise.resolve(response.data);
-    }
-    return Promise.reject(response.data);
+    if (response.status !== 200) {
+      throw new Error('Failed to move chat to anonymous');
+    }
+    return response.data;
```

```diff
-    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
-    return Promise.reject(e);
+    console.error(e);
+    if (e?.response?.data?.message) {
+      throw new Error(e.response.data.message);
+    }
+    throw e;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->